data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -94,6 +95,12 @@ protected DeleteFilter(String filePath, List<DeleteFile> deletes, Schema tableSc
     this.eqDeletes = eqDeleteBuilder.build();
     this.requiredSchema = fileProjection(tableSchema, requestedSchema, posDeletes, eqDeletes);
     this.posAccessor = requiredSchema.accessorForField(MetadataColumns.ROW_POSITION.fieldId());
+    this.hasColumnIsDeleted = requestedSchema.findField(MetadataColumns.IS_DELETED.fieldId()) != null;
+    this.columnIsDeletedIndex = requestedSchema.idToIndex(MetadataColumns.IS_DELETED.fieldId());
```

Comment:
if we rename `idToIndex` to `idToPosition` as suggested in another comment, we can also rename this variable.